src/optimal_control.jl

```diff
@@ -28,17 +30,19 @@ function append(traj::DiscreteTrajectory, t)
     DiscreteTrajectory(traj.q_0, [traj.transitions; t])
 end
 
-struct ContinuousTrajectory{T, VT<:AbstractVector{T}}
-    x::Vector{VT}
-    u::Vector{VT}
+# `x` is a sequence of points in the state space and `u` is a sequence of points
+# in the input space
+struct ContinuousTrajectory{T, XVT<:AbstractVector{T}, UVT<:AbstractVector{T}}
```

Comment:
Good point, if we use `StaticArrays`, the input and state might have different types